argoverse/data_loading/argoverse_stereo_loader.py

```diff
@@ -0,0 +1,432 @@
+        self._image_timestamp_list: Optional[Dict[str, Dict[str, List[int]]]] = None
+        self._timestamp_image_dict: Optional[Dict[str, Dict[str, Dict[int, str]]]] = None
+        self._calib: Optional[Dict[str, Dict[str, Calibration]]] = None  # { log_name: { camera_name: Calibration } }
+        self.counter: int = 0
+
+        self.image_count: int = 0
+
+        self.root_dir: str = root_dir
+        self.split_name: str = split_name
+
+        self.current_log = self.log_list[self.counter]
+
+        assert self.image_list is not None
+
+        # load calibration file
+        self.calib_filename: str = os.path.join(
+            self.root_dir,
+            "rectified_stereo_images_v1.1",
+            self.split_name,
+            self.current_log,
+            "vehicle_calibration_stereo_info.json",
+        )
+
+        # stereo camera @5hz
+        self.num_stereo_camera_frame: int = len(self.image_timestamp_list[STEREO_FRONT_LEFT_RECT])
+
+        assert self.calib is not None
+
+    @property
+    def calib(self) -> Dict[str, Calibration]:
+        """get calibration dict for current log
+
+        Returns:
+            calib: Calibration object for the current log
+        """
+        self._ensure_calib_is_populated()
+        assert self._calib is not None
+        return self._calib[self.current_log]
+
+    def _ensure_calib_is_populated(self) -> None:
+        """load up calibration object for all logs
+
+        Returns:
+            None
+        """
+        if self._calib is None:
+            self._calib = {}
+            for log in self.log_list:
+                calib_filename = os.path.join(
+                    self.root_dir,
+                    "rectified_stereo_images_v1.1",
+                    self.split_name,
+                    log,
+                    "vehicle_calibration_stereo_info.json",
+                )
+                self._calib[log] = load_calib(calib_filename)
+
+    @property
+    def log_list(self) -> List[str]:
+        """return list of log (str) in the current dataset directory
+
+        Returns:
+            log_list: list of string representing log id
+        """
+        if self._log_list is None:
+
+            def valid_log(log: str) -> bool:
+                return os.path.exists(
+                    os.path.join(
+                        self.root_dir,
+                        "rectified_stereo_images_v1.1",
+                        self.split_name,
+                        log,
+                        "vehicle_calibration_stereo_info.json",
+                    )
+                )
+
+            self._log_list = [
+                x
+                for x in os.listdir(os.path.join(self.root_dir, "rectified_stereo_images_v1.1", self.split_name))
+                if valid_log(x)
+            ]
+
+        return self._log_list
+
+    @property
+    def image_list(self) -> Dict[str, List[str]]:
+        """return list of all image path (str) for the current log
```

Comment:
is the doc string correct here -- we loop over `self.log_list` but the docstring says its just for the current log

Reply:
It's right. It loops if the list is empty, then fills it and returns the disparity list for the current log_id. Similar to the argoverse_tracking_loader.